### PR TITLE
Upgrade vr-imgui

### LINKED ISSUE #49: Go up on transfer function widget
The logic doesnt really make sense. the [original repo of this widget](https://github.com/Twinklebear/imgui-transfer-function) changed the implementation about 2 years ago. Maybe try to update it with their latest. 
There is one issue. The transfer function implementation is tested against imgui 1.67. The volume viewer is on imgui 1.83. 
I am opening an issue to ask if the widget is compatible with later versions of imgiu.

https://github.com/Twinklebear/imgui-transfer-function/issues/3

It seems like it's adjusting things so the right side of the equation is always to the right of the `=`, even on multiple lines. Not sure I agree with that decision, especially when we're dealing with long variable names

_Originally posted by @RobertGemmaJr in https://github.com/brown-ccv/VR-Volumeviewer/pull/45#discussion_r864895737_

### PR DESCRIPTION
**Pull request by kmilo9999**
This PR includes linking against the new vr-imgui that supports the latest imgui and the upgrading in transfer function widget.
Closes #49 and #48

### CODE REVIEW

File: libs/UIHelpers/transfer_function_widget.cpp [hunk 1]
```diff
@@ -37,547 +15,525 @@
...
-    return min;
+    if (x < min)
+    {
+      return min;
+    }
+    if (x > max)
+    {
+      return max;
+    }
+    return x;
```
Comment (RobertGemmaJr) [inline libs/UIHelpers/transfer_function_widget.cpp:33]: Are you adding any extra functionality here? It's a little hard to tell in GitHub, just want to make sure you're not merging any changes you don't mean to

File: libs/UIHelpers/transfer_function_widget.cpp [hunk 2]
```diff
@@ -37,547 +15,525 @@
...
+                        { return a.x < b.x; });
+              if (selected_point != 0 && selected_point != alpha_control_pts.size() - 1)
+              {
+                fnd = std::find_if(
+                    alpha_control_pts.begin(), alpha_control_pts.end(), [&](const vec2f &p)
+                    {
+								const vec2f pt_pos = p * view_scale + view_offset;
+								float dist = (pt_pos - vec2f(clipped_mouse_pos)).length();
+								return dist <= point_radius; });
+              }
```
Comment (RobertGemmaJr) [inline libs/UIHelpers/transfer_function_widget.cpp:233]: Run the linter?

**Review by RobertGemmaJr — COMMENTED** [review]
There seems to be a lot more going on here than just updating a version? Is this all for compatibility reasons?

File: libs/UIHelpers/transfer_function_widget.cpp (hunk 2 above)
Comment (kmilo9999) [reply libs/UIHelpers/transfer_function_widget.cpp:233]: Done!

File: libs/UIHelpers/transfer_function_widget.cpp (hunk 1 above)
Comment (kmilo9999) [reply libs/UIHelpers/transfer_function_widget.cpp:33]: Yes. Kind of new functionality. Updating imgui also implies upgrading the code in the transfer function widgets.

**Comment by kmilo9999** [comment]
@RobertGemmaJr  Sorry to be so late on this PR. Could you give it a new review?

**Review by RobertGemmaJr — APPROVED** [review]
Looks like the only changes since last time were or CMake. Looks great!

**Pull request merged**